indico/modules/rb/controllers/user/rooms.py

```diff
@@ -156,19 +158,41 @@ def _checkParams(self):
         self._room = Room.get(request.view_args['roomID'])
         self._occupancy_period = request.args.get('period', 'pastmonth')
         self._end = date.today()
+        self._year = None
+        self._month = None
         if self._occupancy_period == 'pastmonth':
             self._end = self._end - relativedelta(days=1)
             self._start = self._end - relativedelta(days=29)
         elif self._occupancy_period == 'thisyear':
             self._start = date(self._end.year, 1, 1)
         elif self._occupancy_period == 'sinceever':
-            self._start = Reservation.query.with_entities(func.min(Reservation.start_dt)).one()[0].date()
+            oldest = (db.session.query(Reservation.start_dt)
+                      .filter_by(room_id=self._room.id)
+                      .order_by(Reservation.start_dt).first()[0])
```

Comment:
This will fail if there are no reservations.

If you use `db.session.query(db.func.min(Reservation.start_dt))` instead you'll always get a row (so you can use `.one()`) and also don't need the `order_by()`